File: 주은/W3/기능개발.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;
queue<pair<int, int>> q;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int>::iterator it;
    for (int i = 0; i < progresses.size(); i++) {
        q.push({progresses[i], i});
    }
    int day = 1;
    int cnt = 1;
    while(!q.empty()) {
        auto[p, th] = q.front();
        if (p+speeds[th]*day >= 100) {
            q.pop();
            cnt++;
        }
        else {
            if (day != 1) {
                answer.push_back(cnt);
                cnt = 1;
            }
            if ((100-p)%speeds[th] == 0) day = (100-p)/speeds[th];
            else day = (100-p)/speeds[th] + 1;
            q.pop();
        }
    }
    answer.push_back(cnt);
    return answer;
}

File: 서연/W3/기능개발.py
import math

def solution(progresses, speeds):
    days_left = []
    for p, s in zip(progresses, speeds):
        day = math.ceil((100 - p) / s)
        days_left.append(day)
    
    answer = []
    current_day = days_left[0]
    count = 0
    
    for day in days_left:
        if day <= current_day:
            count += 1
        else:
            answer.append(count)
            current_day = day
            count = 1
            
    answer.append(count)
    
    return answer
